packages/block-library/src/post-title/edit.js

```diff
@@ -1,3 +1,17 @@
-export default function PostTitleEdit() {
-	return <h2>{ 'Hello world!' }</h2>;
+/**
+ * WordPress dependencies
+ */
+import { useSelect } from '@wordpress/data';
+
+export default function PostTitleEdit( { context } ) {
+	const { postType, postId } = context;
+	const post = useSelect( ( select ) =>
+		select( 'core' ).getEditedEntityRecord( 'postType', postType, postId )
+	);
+
+	if ( ! post ) {
+		return null;
+	}
+
+	return <h2>{ post.title }</h2>;
```

Comment:
> This will escape any HTML entities inside the post title which, technically, WordPress allows.

Fair point. But for the purposes of this pull request, I'm not especially concerned that this block be very refined. I'd be happy to create an issue so that this is tracked, however.

Reply:
> > This will escape any HTML entities inside the post title which, technically, WordPress allows.
> 
> Fair point. But for the purposes of this pull request, I'm not especially concerned that this block be very refined. I'd be happy to create an issue so that this is tracked, however.

Also, further along this point, the editable representation of this block will probably enable the user to change the title from the block itself, which may require a different solution altogether than to simply render a `<h2>` (e.g. `RichText`). This was included in #19572, but was omitted here for simplicity's sake.

Reply:
> This will escape any HTML entities inside the post title which, technically, WordPress allows.

Tracking: #21663